engine/cdn/storage/types.go

```diff
@@ -284,6 +285,27 @@ func (x RunningStorageUnits) GetBuffer(bufferType sdk.CDNItemType) BufferUnit {
 	}
 }
 
+func (x *RunningStorageUnits) FilterItemUnitReaderByType(ius []sdk.CDNItemUnit) []sdk.CDNItemUnit {
+	// Remove cds backend from getting something that is not a log
+	if ius[0].Type != sdk.CDNTypeItemStepLog && ius[0].Type != sdk.CDNTypeItemServiceLog {
+		var cdsBackendID string
+		for _, unit := range x.Storages {
+			if unit.GetDriverName() == "cds" {
+				cdsBackendID = unit.ID()
+				break
+			}
+		}
+
+		for i, s := range ius {
+			if s.UnitID == cdsBackendID {
+				ius = append(ius[:i], ius[i+1:]...)
```

Comment:
Not sure that it works if i == len(ius)-1

Reply:
it works